db/migrations/mysql/000083_threads_threaddeleteat.up.sql

```diff
@@ -0,0 +1,35 @@
+-- Drop any existing DeleteAt column from 000081_threads_deleteat.up.sql
+SET @preparedStatement = (SELECT IF(
+    EXISTS(
+        SELECT 1 FROM INFORMATION_SCHEMA.STATISTICS
+        WHERE table_name = 'Threads'
+        AND table_schema = DATABASE()
+        AND column_name = 'DeleteAt'
+    ) > 0,
+    'ALTER TABLE Threads DROP COLUMN DeleteAt;',
+    'SELECT 1;'
+));
+
+PREPARE removeColumnIfExists FROM @preparedStatement;
+EXECUTE removeColumnIfExists;
+DEALLOCATE PREPARE removeColumnIfExists;
+
+SET @preparedStatement = (SELECT IF(
+    NOT EXISTS(
+        SELECT 1 FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Threads'
+        AND table_schema = DATABASE()
+        AND column_name = 'ThreadDeleteAt'
+    ),
+    'ALTER TABLE Threads ADD COLUMN ThreadDeleteAt bigint(20);',
+    'SELECT 1;'
+));
+
+PREPARE addColumnIfNotExists FROM @preparedStatement;
+EXECUTE addColumnIfNotExists;
+DEALLOCATE PREPARE addColumnIfNotExists;
+
+UPDATE Threads, Posts
+SET Threads.ThreadDeleteAt = Posts.DeleteAt
+WHERE Posts.Id = Threads.PostId
+AND Threads.ThreadDeleteAt IS NULL;
```

Comment:
We generally try to guard the update statements with `IF` condition if applicable. In this case I guess `WHERE Threads.ThreadDeleteAt IS NULL` should serve this purpose right?

Reply:
Ah, this is an interesting nuance. My change effectively migrates bad data going forward, whereas an IF check does it once. But the IF check has a known "cost", whereas even doing nothing in this query might take time on each startup.

But how do we do that? In practice, a data only migration would need "another bit" to remember if we've already done that, seeing as we run them every time. Do we have an example of this? (I guess in this case we could key off whether the column existed at the start of the migration.)

Reply:
For this PR, I'm inclined to leave things as-is as you note above -- since the cost is the same for both IF checks.

But I wonder if we might start a conversation around /not/ running migrations every time? It might be conceptually easier to reason about if the default behaviour was to run the migration once, for cases like this.

Reply:
Yeah, we run the migrations only once if we already successfully complete it. My concern was for edge cases where we need to run multiple times (in case of errors).

Reply:
> Yeah, we run the migrations only once if we already successfully complete it. My concern was for edge cases where we need to run multiple times (in case of errors).

Oh! I misunderstood this then! Yeah, if we already gate it at "run once", then any overhead here seems exceptional, and the `WHERE` definitely avoids writing when unnecessary.

Reply:
Thanks for the thorough review @isacikgoz !